### PR TITLE
Issue #3375: When a theme has a managed_file field and a submit callback, call to undefined function error is thrown

### PR DESCRIPTION
**Pull request by klonos**
http://drupal.org/node/2900373

>**Opportunity overview**:
>This bug is preventing many themes from working with managed files. Most notably Template monster is the most affected, dozens of quality themes break when updated because of this core issue. Many themers have pulled out all their hairs and given up along the way. For an average themer, Drupalista, or even an expert awesome Drupalista it is not easy to figure out why this is happening so lets just fix it so that we can restore confidence in our product.
>
>Here is an example of some very awesome themes are affected and REQUIRE this issue to be fixed:
>http://info.template-help.com/help/drupal-template.html

### CODE REVIEW

File: core/modules/system/tests/system.test [hunk 1]
```diff
@@ -1648,6 +1648,29 @@ class SystemThemeFunctionalTest extends BackdropWebTestCase {
...
+    // Enable the test theme and the module that controls it. Clear caches in
+    // between so that the module's hook_system_theme_info() implementation is
+    // correctly registered.
+    module_enable(array('theme_test'));
+    backdrop_flush_all_caches();
+    theme_enable(array('test_theme'));
+
+    // Test that the theme-specific settings form can be saved and that the
+    // theme-specific checkbox is checked and unchecked as appropriate.
+    $this->drupalGet('admin/appearance/settings/test_theme');
```
Comment (herbdool) [inline core/modules/system/tests/system.test]: should be `backdropGet`

File: core/modules/system/tests/system.test [hunk 2]
```diff
@@ -1648,6 +1648,29 @@ class SystemThemeFunctionalTest extends BackdropWebTestCase {
...
+    $this->assertNoFieldChecked('edit-test-theme-checkbox', 'The test_theme_checkbox setting is unchecked.');
+    $this->drupalPost(NULL, array('test_theme_checkbox' => TRUE), t('Save theme settings'));
```
Comment (herbdool) [inline core/modules/system/tests/system.test]: `backdropPost`

File: core/modules/system/tests/system.test [hunk 3]
```diff
@@ -1648,6 +1648,29 @@ class SystemThemeFunctionalTest extends BackdropWebTestCase {
...
+    $this->assertText('The configuration options have been saved.');
+    $this->assertFieldChecked('edit-test-theme-checkbox', 'The test_theme_checkbox setting is checked.');
+    $this->drupalPost(NULL, array('test_theme_checkbox' => FALSE), t('Save theme settings'));
```
Comment (herbdool) [inline core/modules/system/tests/system.test]: `backdropPost`

File: core/modules/system/tests/system.test (hunk 1 above)
Comment (klonos) [reply core/modules/system/tests/system.test]: Good catch @herbdool 👍 ...fixed

File: core/modules/system/tests/system.test (hunk 2 above)
Comment (klonos) [reply core/modules/system/tests/system.test]: Fixed 😉

File: core/modules/system/tests/system.test (hunk 3 above)
Comment (klonos) [reply core/modules/system/tests/system.test]: ...and also fixed 😉

**Comment by herbdool** [comment]
Tests not passing.

**Comment by klonos** [comment]
I have tried restarting the tests, but the php5 test would not fire up for some reason. Lets see now.

**Comment by klonos** [comment]
Tests are now green @herbdool 😉

File: core/modules/simpletest/tests/themes/test_theme/theme-settings.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?php
+ /**
+ * @file
+ * Theme settings file for the test_theme theme.
+ */
+ /**
+ * Implements hook_form_FORM_ID_alter().
```
Comment (herbdool) [inline core/modules/simpletest/tests/themes/test_theme/theme-settings.php]: Don't need this comment anymore.

**Comment by herbdool** [comment]
I'm curious about the change in theme-settings.php. Now it looks right but why was there a form alter there before?

File: core/modules/simpletest/tests/themes/test_theme/theme-settings.php (hunk 4 above)
Comment (klonos) [reply core/modules/simpletest/tests/themes/test_theme/theme-settings.php]: Indeed. Removed.

**Comment by klonos** [comment]
> I'm curious about the change in theme-settings.php. Now it looks right but why was there a form alter there before?

I guess that this is how `theme-settings.php` files are implemented in D7. See https://github.com/drupal/drupal/blob/7.x/themes/garland/theme-settings.php#L16 for example

**Comment by herbdool** [comment]
Interesting. I didn't realize that changed.

**Comment by herbdool** [comment]
RTBC

**Comment by klonos** [comment]
> Interesting. I didn't realize that changed.

Neither did I 😄 ...see https://github.com/backdrop/backdrop-issues/issues/3391

**Comment by quicksketch** [comment]
Thanks! Merged into 1.x and 1.11.x.

**Pull request merged**